### DISCUSSION TITLE
Switch to PER Coding Style 2.0

### DISCUSSION THREAD

**Discussion by shyim** [shopware/platform#3420] [RFC / Proposal]
We use right now PSR-12 with a lot of custom rules for our code base. To simplify the code style config, we should migrate to [PER Coding Style 2.0](https://www.php-fig.org/per/coding-style/) as it is a successor to PSR-12.

Some mentionable changes to the current code-style:

## Attribute is newline to property:
![image](https://github.com/shopware/shopware/assets/6224096/fe4ae358-396e-47e6-9e73-d9b3715bcf7c)

## Empty method/function body are closed in the same line:

![image](https://github.com/shopware/shopware/assets/6224096/79f737bc-39ba-4b43-99e0-78f3db17df67)

## New line is required after start PHP tag:

![image](https://github.com/shopware/shopware/assets/6224096/76e8b472-ec4a-4da0-b1e0-c3328d02dab2)

**Comment by mitelg** [comment]
do we want to remove all the custom rules, too?

**Comment by lasomethingsomething** [comment]
@shyim Checking in on this—something you still advocate us doing?

**Reply by shyim** [reply]
I think the state right now is fine
